pkgs/by-name/li/libpostal/package.nix

```diff
@@ -1,34 +1,44 @@
-{ lib, stdenv, fetchFromGitHub, fetchpatch, autoreconfHook }:
+{ lib, stdenv, fetchFromGitHub, fetchurl, autoreconfHook, withData ? false }:
 
-stdenv.mkDerivation rec {
+let
+  releases = "https://github.com/openvenues/libpostal/releases";
+  assets-base = fetchurl {
+    url = "${releases}/download/v1.0.0/libpostal_data.tar.gz";
+    hash = "sha256-0uxQWHvzp+RuGOXc3jJBkTQmb5B3TjlW8sL5DYGP+aE=";
+  };
+  assets-parser = fetchurl {
+    url = "${releases}/download/v1.0.0/parser.tar.gz";
+    hash = "sha256-cZTpsAlfca7LhhJp9nXlBwPnPjUqC51Bxg+NjvWgNiQ=";
+  };
+  assets-language-classifier = fetchurl {
+    url = "${releases}/download/v1.0.0/language_classifier.tar.gz";
+    hash = "sha256-FqbsttN+fiXY/lFCVWZoUqudvU2cx2L2TPHhW0Nponc=";
+  };
+in stdenv.mkDerivation rec {
   pname = "libpostal";
   version = "1.1";
 
   src = fetchFromGitHub {
     owner = "openvenues";
     repo = "libpostal";
-    rev = "v${version}";
-    sha256 = "sha256-gQTD2LQibaB2TK0SbzoILAljAGExURvDcF3C/TfDXqk=";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-7G/CjYdVzsrvUFXGODoXgXoRp8txkl5SddcPtgltrjY=";
   };
 
-  patches = [
-    # Fix darwin compilation with XCode 12 https://github.com/openvenues/libpostal/issues/511
-    (fetchpatch {
-      name = "Fix-C-compilation-macOS.patch";
-      url = "https://github.com/openvenues/libpostal/commit/9fcf066e38121b5c1439fc6bdc9a7e02234c8622.patch";
-      hash = "sha256-VpboGK+5sc1XrxMB051KWc8vP7Eu2g7zmTirzSaerns=";
-    })
-    # https://github.com/openvenues/libpostal/commit/bfdb6b8f87cc1cae9ba47870ff23deae0bb8ba51.patch
-    # with extra hunk removed so it applies
-    ./0001-test-adding-header-to-fix-warning.patch
-  ];
-
   nativeBuildInputs = [ autoreconfHook ];
 
   configureFlags = [
     "--disable-data-download"
   ] ++ lib.optionals stdenv.hostPlatform.isAarch64 [ "--disable-sse2" ];
 
+  postBuild = lib.optionalString withData ''
+    mkdir -p $out/share/libpostal
+    tar -C $out/share/libpostal -xvf ${assets-base}
```

Comment:
The other PR used `fetchzip` instead of `fetchurl` for the data assets, and then simply symlinked them into the libpostal output.

It's not a big deal, but I think that might result save a bit of space if libpostal gets updated or otherwise modified but the data doesn't (which is fairly likely: the data is basically a fixed-output derivation, so it should only change if its version changes, which doesn't happen often: it's still on v1.0.0. By contrast, libpostal will be rebuilt whenever it or its dependencies change). So for example, if we have two builds of libpostal using the same data assets, this scheme using fetchurl and tar will copy the data into each libpostal output and possibly garbage-collect and re-download the original tarballs each time, whereas using fetchzip and symlinking should result in both libpostal outputs depending on the same data asset outputs.

`nix store optimise` should be able to deduplicate the data, but unless there's a good reason not to, it's probably better not to duplicate it in the first place.

Reply:
Done, updated.